### ISSUE TITLE
Bug: [no-unused-vars] no `is assigned a value but only used as a type` error when it has a same name type alias declaration exported

### ISSUE THREAD

**Issue by Dunqing** [typescript-eslint/typescript-eslint#10658]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.7.2&fileType=.ts&code=MYewdgzgLgBAgjAvDADAbgFAYKYA8AOIATrFAJ77bxIzmUgBm8aQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQHYHsBaWXWZRAE0IDcBDaVDSfAMxcgBpxtIABAFwCeABxQBjaAEth-QiniTc-APQFipclToN0URNGj5okbgF8QpoA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
const A = 0;
export type A = typeof A;
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  {
   "rules": {
    "no-unused-vars": "off",
    "@typescript-eslint/no-unused-vars": "error"
    }
  }
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

Should report an error same as https://typescript-eslint.io/play/#ts=5.7.2&fileType=.ts&code=MYewdgzgLgBAgjAvDADAbgFAagTwA4Cm8SMuhIAZvGkA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQHYHsBaWXWZRAE0IDcBDaVDSfAMxcgBpxtIABAFwCeABxQBjaAEth-QiniTc-APQFipclToN0URNGj5okbgF8QpoA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

```ts
const A = 0;
type A = typeof A;
```

### Actual Result

No error reported

### Additional Info

Related: https://github.com/oxc-project/oxc/issues/8488#issuecomment-2589932781

**Comment by bradzacher** [comment]
it's a very, very rare edge case, but it should be reported.

**Comment by Dunqing** [comment]
> it's a very, very rare edge case, but it should be reported.

Thanks for confirming this!
